### PR TITLE
Add basic prototype

### PR DESCRIPTION
**Pull request by tcode2k16**

### CODE REVIEW

**Review by ekilmer — CHANGES_REQUESTED** [review]
Could you also update the README, please

File: mui/mui.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        block.function.set_auto_instr_highlight(addr, HighlightStandardColor.RedHighlightColor)
+
+    # Add the instruction to the list associated with the current view
+    bv.session_data.mui_avoid.add(addr)
+
+
+def solve(bv):
+    if len(bv.session_data.mui_find) == 0:
+        show_message_box(
+            "Angr Solve",
```
Comment (ekilmer) [inline mui/mui.py:84]:
```suggestion
            "Manticore Solve",
```

**Review by ekilmer — CHANGES_REQUESTED** [review]
Looking good! Would be nice to have some docstrings and more typing for function arguments.

**Comment by tcode2k16** [comment]
I made some more changes based on feedback and tested the code on both mac and linux. This pull request should now meet all acceptance criteria for #1

File: mui/mui.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+
+class ManticoreRunner(BackgroundTaskThread):
+    def __init__(self, find: Set[int], avoid: Set[int], view: BinaryView):
+        BackgroundTaskThread.__init__(self, "Solving with manticore...", True)
+        self.find = tuple(find)
+        self.avoid = tuple(avoid)
+        self.view = view
+
+        # Write the binary to disk so that the angr API can read it
```
Comment (ekilmer) [inline mui/mui.py:28]:
```suggestion
        # Write the binary to disk so that the Manticore API can read it
```

File: mui/mui.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+from manticore.core.state import StateBase
+from manticore.native import Manticore
+
+BinaryView.set_default_session_data("mui_find", set())
+BinaryView.set_default_session_data("mui_avoid", set())
+
+
+class ManticoreRunner(BackgroundTaskThread):
+    def __init__(self, find: Set[int], avoid: Set[int], view: BinaryView):
+        BackgroundTaskThread.__init__(self, "Solving with manticore...", True)
```
Comment (ekilmer) [inline mui/mui.py:23]:
```suggestion
        BackgroundTaskThread.__init__(self, "Solving with Manticore...", True)
```

File: mui/mui.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        self.find = tuple(find)
+        self.avoid = tuple(avoid)
```
Comment (ekilmer) [inline mui/mui.py:25]: Good catch for making these collections immutable when handling them with Manticore 👍  💯

File: requirements.txt [hunk 5]
```diff
@@ -0,0 +1 @@
+-e './manticore[native]'
```
[the remark this answers is not in the document]
Comment (tcode2k16) [reply requirements.txt:1]: Yes, I'm able to get autocompletion and type hints working in PyCharm by including paths to Binary Ninja's Python folders. I'm not sure how much typing information `mypy` is able to use, but thus far I have not run into any issues with that.

**Review by ekilmer — APPROVED** [review]
Great! LGTM

File: requirements.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (sschriner) [reply requirements.txt:1]: My issue was derived in the fact I had not switched to the development version, so this suggestion is irrelevant

**Pull request merged**